cpp/src/sampling/detail/gather_one_hop_impl.cuh

```diff
@@ -526,97 +497,101 @@ temporal_gather_one_hop_edgelist(
+                                edge_times.begin());
+
+    auto [keep_count, marked_entries] =
+      tmp_positions
+        ? detail::mark_entries(handle,
+                               edge_times.size(),
+                               [d_tmp           = edge_times.data(),
+                                d_tmp_positions = tmp_positions->data(),
+                                kv_store_view =
+                                  kv_binary_search_store_device_view_t<decltype(kv_store.view())>{
+                                    kv_store.view()}] __device__(auto index) {
+                                 auto edge_time = d_tmp[index];
+                                 auto key_time =
+                                   cuda::std::get<0>(kv_store_view.find(d_tmp_positions[index]));
+                                 return (edge_time > key_time);
+                               })
+        : detail::mark_entries(
+            handle,
+            edge_times.size(),
+            [d_tmp = edge_times.data(), d_tmp_time = tmp_times->data()] __device__(auto index) {
+              auto edge_time = d_tmp[index];
+              auto key_time  = d_tmp_time[index];
+              return (edge_time > key_time);
+            });
+
+    raft::device_span<uint32_t const> marked_entry_span{marked_entries.data(),
+                                                        marked_entries.size()};
+
+    result_srcs =
+      detail::keep_marked_entries(handle, std::move(result_srcs), marked_entry_span, keep_count);
+    result_dsts =
+      detail::keep_marked_entries(handle, std::move(result_dsts), marked_entry_span, keep_count);
+    if (std::holds_alternative<rmm::device_uvector<edge_t>>(tmp_edge_indices)) {
+      tmp_edge_indices = detail::keep_marked_entries(
+        handle,
+        std::move(std::get<rmm::device_uvector<edge_t>>(tmp_edge_indices)),
+        marked_entry_span,
+        keep_count);
+    }
+    if (tmp_times) {
+      *tmp_times =
+        detail::keep_marked_entries(handle, std::move(*tmp_times), marked_entry_span, keep_count);
+    }
+    if (tmp_positions) {
+      *tmp_positions = detail::keep_marked_entries(
+        handle, std::move(*tmp_positions), marked_entry_span, keep_count);
+    }
 
-  edge_list.clear();
-  edge_list.insert(result_srcs.begin(),
-                   result_srcs.end(),
-                   result_dsts.begin(),
-                   tmp_edge_indices ? std::optional<edge_t*>{tmp_edge_indices->begin()}
-                                    : std::optional<edge_t*>{std::nullopt});
+    result_labels = rmm::device_uvector<label_t>(keep_count, handle.get_stream());
+    thrust::transform(
+      handle.get_thrust_policy(),
+      tmp_positions->begin(),
+      tmp_positions->end(),
+      result_labels->begin(),
+      [kv_store_view = kv_binary_search_store_device_view_t<decltype(kv_store.view())>{
+         kv_store.view()}] __device__(auto pos) {
+        return cuda::std::get<1>(kv_store_view.find(pos));
+      });
```

Comment:
Why not just call kv_store_view.find()? (https://github.com/rapidsai/cugraph/blob/branch-25.08/cpp/src/prims/kv_store.cuh#L257).